unstructured/partition/auto.py

```diff
@@ -524,6 +541,13 @@ def partition(
             detect_language_per_element=detect_language_per_element,
             **kwargs,
         )
+    elif filetype in CODE_FILETYPES:
```

Comment:
Could we update this logic to fallback to `partition_text` if `tree_sitter` doesn't exist? That way people who a processing these as text files can still do so if they don't have the code dependencies installed.

Reply:
Sure, 
We can either :
- define `partition_code = partition_text` when checking for the dependancy
- Add a global switch (or recheck for dependancy) and change the filetype from any code filetype to TXT
- Add a global switch (or recheck for dependancy) and add a nested if in the `if filetype in CODE_FILETYPES:`

I'm fine with any of those

Reply:
Added tests, 
There are some gotcha that I will try to fix such as javascript files detected as cpp by libmagic, or go file detected as plain text

On top of that, I found that it's tricky to make the text actually fit both the minimum and the maximum size properly. I went with enforcing the minimum because it means less elements so potentially better search in the vector database

Reply:
I'm good with either of these. I'd say avoid `partition_code = partition_text`, just so it's clear later in the code that `partition_text` is being applied and not `partition_code`.
- Add a global switch (or recheck for dependancy) and change the filetype from any code filetype to TXT
- Add a global switch (or recheck for dependancy) and add a nested if in the if filetype in CODE_FILETYPES:

Are the MIME type discrepancies when they're checked as files rather than by filename? If think that's okay and you can mock the correct MIME type in the test. Hopefully most of those files come in with the file extension.

The enforcing minimum logic makes sense to me.

Reply:
Yes, MIME types are not always correctly detected.
Just to make sure people have the option, I'm adding the `programming_language` parameters, and adding some test for the auto partition 

We could try all the available parsers for file that are not detected as code at all, but for the ones where the MIME is mixed up, there's nothing we can really do

Reply:
After looking up a bit, it seems that most code files do not have a MIME type...
Here is a list of available ones https://www.iana.org/assignments/media-types/media-types.xhtml

Since it is required for the auto partition, I made custom ones, it looks like to me that the common way of doing so is `text/x-{language}` so I went for that